src/rust/catmem/futures/push.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//======================================================================================================================
+// Imports
+//======================================================================================================================
+
+use crate::{
+    catmem::SharedRingBuffer,
+    runtime::{
+        fail::Fail,
+        memory::DemiBuffer,
+    },
+    QDesc,
+};
+use ::std::{
+    future::Future,
+    pin::Pin,
+    rc::Rc,
+    task::{
+        Context,
+        Poll,
+    },
+};
+
+//======================================================================================================================
+// Structures
+//======================================================================================================================
+
+/// Push Operation Descriptor
+pub struct PushFuture {
+    /// Associated queue descriptor.
+    qd: QDesc,
+    idx: usize,
```

Comment:
Assuming idx is an index, can we spell it out please?

Reply:
Sure.